pkg/controller/gslb/metrics.go

```diff
@@ -0,0 +1,72 @@
+package gslb
+
+import (
+	ohmyglbv1beta1 "github.com/AbsaOSS/ohmyglb/pkg/apis/ohmyglb/v1beta1"
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	ohmyglbNamespace = "ohmyglb"
+	gslbSubsystem    = "gslb"
+)
+const (
+	healthyStatus   = "Healthy"
+	unhealthyStatus = "Unhealthy"
+	notFoundStatus  = "NotFound"
+)
+
+// Custom gslb prometheus metrics
+var (
+	healthyRecordsMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: ohmyglbNamespace,
+			Subsystem: gslbSubsystem,
+			Name:      "healthy_records",
+			Help:      "Number of healthy records observed by OhMyGLB.",
+		},
+		[]string{"namespace", "name"},
+	)
+	ingressHostsPerStatusMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: ohmyglbNamespace,
+			Subsystem: gslbSubsystem,
+			Name:      "ingress_hosts_per_status",
+			Help:      "Number of managed hosts observed by OhMyGLB.",
+		},
+		[]string{"namespace", "name", "status"},
+	)
+)
+
+func (r *ReconcileGslb) updateManagedHostsTotalMetric(gslb *ohmyglbv1beta1.Gslb, serviceHealth map[string]string) error {
+	var healthyTotal, unhealthyTotal, notFoundTotal int
+	for _, hs := range serviceHealth {
+		switch hs {
+		case healthyStatus:
+			healthyTotal++
```

Comment:
@somaritane just a note that internal variable do not comply with upstream Total suffix semantics, do I get it right ?

Reply:
Well spotted, @ytsarev! Makes sense to rename method as well.